### ISSUE TITLE
Cannot get first record with where condition

### ISSUE THREAD

**Issue by jimmymaise** [betodealmeida/shillelagh#198]
```
from shillelagh.backends.apsw.db import connect

connection = connect(":memory:")
cursor = connection.cursor()

query = "SELECT * FROM 'https://docs.google.com/spreadsheets/d/1g1ICysVgR5TkRv6nWTgJIFCH0ki3aj0vV_62yXICqjM/edit#gid=581871282'" \
        "where id = '5c47428c-8dc2-4fe7-acfd-7d06ba746ac9'"
for row in cursor.execute(query):
    print(row)
```

In the below code, I expect to get two rows but actually just one- It's the second row. 

If I select *, I can get two row. But with where condition, I just can get only one row even though both two rows match where condition
```

**Comment by betodealmeida** [comment]
Ah, good catch. This is actually a bug in how we handle sheets where the column names are not identified correctly (for some reason, in your sheet above the API things the first row is data). Let me fix it.

In the meantime, you can append `?header=1` to the URL to make it work: https://docs.google.com/spreadsheets/d/1g1ICysVgR5TkRv6nWTgJIFCH0ki3aj0vV_62yXICqjM/edit?headers=1#gid=581871282

**Comment by betodealmeida** [comment]
@jimmymaise I just published 1.0.9 with a fix.
